docs/en/reference/architecture.rst

```diff
@@ -1,50 +1,41 @@
 Architecture
 ============
 
-As already said, the DBAL is a thin layer on top of PDO. PDO itself
-is mainly defined in terms of 2 classes: ``PDO`` and
-``PDOStatement``. The equivalent classes in the DBAL are
-``Doctrine\DBAL\Connection`` and ``Doctrine\DBAL\Statement``. A
-``Doctrine\DBAL\Connection`` wraps a
-``Doctrine\DBAL\Driver\Connection`` and a
-``Doctrine\DBAL\Statement`` wraps a
-``Doctrine\DBAL\Driver\Statement``.
-
-``Doctrine\DBAL\Driver\Connection`` and
-``Doctrine\DBAL\Driver\Statement`` are just interfaces. These
-interfaces are implemented by concrete drivers. For all PDO based
-drivers, ``PDO`` and ``PDOStatement`` are the implementations of
-these interfaces. Thus, for PDO-based drivers, a
-``Doctrine\DBAL\Connection`` wraps a ``PDO`` instance and a
-``Doctrine\DBAL\Statement`` wraps a ``PDOStatement`` instance.
-
-What does a ``Doctrine\DBAL\Connection`` or a
-``Doctrine\DBAL\Statement`` add to the underlying driver
+The DBAL consists of two layers: drivers and a wrapper. Each layer
+is mainly defined in terms of 3 components: ``Connection``,
+```Statement`` and ``Result``.
+A ``Doctrine\DBAL\Connection`` wraps a ``Doctrine\DBAL\Driver\Connection``,
+a ``Doctrine\DBAL\Statement`` wraps a ``Doctrine\DBAL\Driver\Statement``
+and a ``Doctrine\DBAL\Result`` wraps a ``Doctrine\DBAL\Driver\Result``.
+
+``Doctrine\DBAL\Driver\Connection``, ``Doctrine\DBAL\Driver\Statement``
+and ``Doctrine\DBAL\Driver\Result`` are just interfaces.
+These interfaces are implemented by concrete drivers.
+
+What do the wrapper components add to the underlying driver
 implementations? The enhancements include SQL logging, events and
 control over the transaction isolation level in a portable manner,
 among others.
 
-A DBAL driver is defined to the outside in terms of 3 interfaces:
-``Doctrine\DBAL\Driver``, ``Doctrine\DBAL\Driver\Connection`` and
-``Doctrine\DBAL\Driver\Statement``. The latter two resemble (a
-subset of) the corresponding PDO API.
-
-A concrete driver implementation must provide implementation
-classes for these 3 interfaces.
+Apart from the three main components, a DBAL driver should also provide
+an implementation of the ``Doctrine\DBAL\Driver`` interface that
+implements the logic of translation of the DBAL connection parameters
+to the ones specific to the given driver and provides factory methods
```

Comment:
Does the interface provide the factory methods? If not no s after provide

Reply:
Yeah, it was about "interface … provides" but it's clumsy and the sentence is too complex. Let me try to rephrase.